client/src/app/gateways/http-stream/http-stream.service.ts

```diff
@@ -93,7 +93,7 @@ export class HttpStreamService {
     }
 
     private shouldReconnect({ error }: ShouldReconnectContext): boolean | Promise<boolean> {
-        if (error.type === PossibleStreamError.AUTH) {
+        if (error.error.type === PossibleStreamError.AUTH) {
```

Comment:
See http-stream.ts `parseCommunicationError` (around ln. 396): I believe the return of this function's result is fed into the `shouldReconnect` function as a parameter. The return has the type `CommunicationError` ( `{type: string, msg: text}` ).
I ofc haven't tested this, but I have a suspicion that your solution in connection with this may cause a TypeError, since by calling error.error you'd be receiving a null value as output.

maybe do `error.error?.type` or smth

Reply:
Good catch! I even intended to do that but seems like I forgot